scipy/stats/tests/test_multivariate.py

```diff
@@ -363,6 +364,86 @@ def test_frozen_dirichlet(self):
             assert_equal(d.pdf(x[:-1]), dirichlet.pdf(x[:-1], alpha))
             assert_equal(d.logpdf(x[:-1]), dirichlet.logpdf(x[:-1], alpha))
 
+    def test_numpy_rvs_shape_compatibility(self):
+        np.random.seed(2846)
+        alpha = np.array([1.0, 2.0, 3.0])
+        x = np.random.dirichlet(alpha, size=7)
+        assert_equal(x.shape, (7, 3))
+        assert_raises(ValueError, dirichlet.pdf, x, alpha)
+        assert_raises(ValueError, dirichlet.logpdf, x, alpha)
+        dirichlet.pdf(x.T, alpha)
+        dirichlet.pdf(x.T[:-1], alpha)
+        dirichlet.logpdf(x.T, alpha)
+        dirichlet.logpdf(x.T[:-1], alpha)
```

Comment:
> >   could the rvs be incompatibly changed like numpy/numpy#5872?

> This sounds like something the random_state arg be useful for.

Better to add a new keyword if needed I think. Overloading ``random_state`` to take inputs other than a seed or ``RandomState`` instance makes it inconsistent with all other ``random_state`` keywords in the rest of the module.